sec-keys/sigstore-trusted-root/sigstore-trusted-root-0_p20241010.ebuild

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="trusted-root.json for dev-python/sigstore"
+HOMEPAGE="https://www.sigstore.dev/"
+SRC_URI="
+	https://dev.gentoo.org/~mgorny/dist/${P}.tar.xz
+	test? (
+		https://www.python.org/ftp/python/3.13.0/Python-3.13.0.tar.xz.sigstore
+	)
+"
+S=${WORKDIR}
+
+LICENSE="public-domain"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="test"
+PROPERTIES="test_network"
+RESTRICT="test"
+
+BDEPEND="
+	test? (
+		dev-python/sigstore
+		sys-apps/diffutils
+	)
+"
+
+src_test() {
+	local common_args=(
+		--bundle "${DISTDIR}"/Python-3.13.0.tar.xz.sigstore
+		--cert-identity thomas@python.org
+		--cert-oidc-issuer https://accounts.google.com
+		sha256:086de5882e3cb310d4dca48457522e2e48018ecd43da9cdf827f6a0759efb07d
```

Comment:
It's fascinating that the tool allows you to verify a sha256sum directly. I assume this is the sum for the python distfile, which makes this a neat hack for avoiding a giant distfile for test purposes.

But it makes me very suspicious that this supposed brand new best cryptographic scheme to obsolete all the rest, chooses a weak hash right out of the gate. I mean, the fact that gnupg lets you use sha512 (much better!!!) but not blake2 is understandable as there's a compatibility spec to deal with and gnupg literally quotes that rationale in their docs.

Greenfielding your own new signing ecosystem and not bothering to support blake2 sure is a flex of some sort, not sure how else to describe it.